addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -366,7 +366,7 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         
         const targetModelName = this.$target[0].dataset.model_name || 'mail.mail';
         this.activeForm = this.models.find(m => m.model === targetModelName);
-        currentActionName = this.activeForm.website_form_label;
+        currentActionName = this.activeForm && this.activeForm.website_form_label;
```

Comment:
Indeed, we should have moved that line inside the `if (!this.modelCantChange)`.
But with your fix, it should also avoid failing for forms where `hide-change-model` has not been specified - until an action is selected.